aineko/extras/connectors/rest_poller.py

```diff
@@ -0,0 +1,260 @@
+    field in `node_params`. If the REST endpoint is not expected to be
+    available at all times, it is recommended to set `max_retries` to a
+    finite number to prevent the node from retrying forever.
+
+    By default, this node will retry connecting to the REST endpoint every 5
+    seconds if the connection fails. This can be changed by setting the
+    `retry_sleep` field in `node_params`. If the REST endpoint is not expected
+    to be available at all times, it is recommended to increase `retry_sleep`
+    to reduce the number of requests sent to the endpoint.
+
+    By default, this node will consider any HTTP status code in the 200s to be
+    a success code. This can be changed by setting the `success_codes` field
+    in `node_params`. If the REST endpoint returns a non-200 status code on
+    success, it is recommended to add the status code to the `success_codes`
+    list.
+
+    By default, this node will not attach headers or data to the request. This
+    can be changed by setting the `headers` and `data` fields in `node_params`.
+    If the REST endpoint requires headers or data to be sent, it is
+    you should set the `headers` and `data` fields in `node_params`.
+
+    By default, this node will timeout after 10 seconds. This can be changed
+    by setting the `timeout` field in `node_params`. If the REST endpoint is
+    expected to take a long time to respond, it is recommended to increase
+    `timeout` to prevent the node from timing out.
+    """
+
+    # Poll settings
+    last_poll_time = time.time()
+    retry_count = 0
+
+    def _pre_loop_hook(self, params: Optional[Dict] = None) -> None:
+        """Initializes connection to API."""
+        # Cast params to ParamsREST type
+        try:
+            if params is not None:
+                self.rest_params = ParamsRESTPoller(**params)
+            else:
+                raise ValueError(
+                    "No params provided to REST connector. "
+                    "Node requires at least a url param."
+                )
+        except Exception as err:  # pylint: disable=broad-except
+            # Cast pydantic validation error to ValueError
+            # so that it can be properly caught by the node
+            # Note: this is required because pydantic errors
+            # are not pickleable
+            raise ValueError(
+                "Failed to cast params to ParamsREST type. "
+                f"The following error occurred: {err}"
+            ) from err
+
+        # Ensure only one output dataset is provided
+        output_datasets = [
+            dataset for dataset in self.producers.keys() if dataset != "logging"
+        ]
+        if len(output_datasets) > 1:
+            raise ValueError(
+                "Only one output dataset is allowed for the "
+                "RESTPoller connector. "
+                f"{len(output_datasets)} datasets given."
+            )
```

Comment:
I don't think this is documented in the docstrings, right? I would definitely include that in the docs.

Reply:
done